### PR TITLE
Unlearning methods + Eval organize

### PR DESCRIPTION
**Pull request by Dornavineeth**
# Done
- [x] Eval Code reorganized
- [x] Unlearning baselines added.

# Unlearning baselines supported
- [x] Gradient Ascent
- [x] GradDiff
- [x] NPO
- [x] IDK DPO
- [ ] SimNPO
- [ ] SOUL
- [ ] Task Vectors

# TODO
- [ ] Experiments folder in configs for each unlearning, fine-tuning experiment
- [ ] Documentation -- Both Readme and Comments in code
- [ ] Logging instead of printing

### CODE REVIEW

File: src/trainer/unlearn/npo.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        self.beta = beta
+
+    def compute_loss(self, model, inputs, return_outputs=False):
+        forget_inputs = inputs["forget"]
+        forget_labels = forget_inputs["labels"]
+        forget_inputs = {
+            "input_ids": forget_inputs["input_ids"],
+            "attention_mask": forget_inputs["attention_mask"],
+        }
+        forget_loss, forget_outputs = compute_dpo_loss(
```
Comment (molereddy) [inline src/trainer/unlearn/npo.py:17]: maybe explicitly write win_input=None
otherwise it will be unclear since default value is unknown to people viewing the file

File: src/trainer/unlearn/base.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+    def evaluate(
+        self,
+        eval_dataset: Optional[Union[Dataset, Dict[str, Dataset]]] = None,
+        ignore_keys: Optional[List[str]] = None,
+        metric_key_prefix: str = "eval",
+    ) -> Dict[str, float]:
+        return super().evaluate(eval_dataset, ignore_keys, metric_key_prefix)
+
+    def prediction_step(
```
Comment (molereddy) [inline src/trainer/unlearn/base.py:105]: If we are copying big amounts of code from any place people usually link to that page and cite

File: configs/trainer/finetune.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 name: finetune
 args:
-  per_device_train_batch_size: 16
+  per_device_train_batch_size: 2
```
Comment (molereddy) [inline configs/trainer/finetune.yaml:3]: change output_dir to default
